recipes/hive_metastore.rb

```diff
@@ -40,6 +40,18 @@
   end
 end
 
+derby_dir =
+  if hive_sql == 'derby'
+    node['hive']['hive_site']['javax.jdo.option.ConnectionURL'][2].split(';').find { |o| /^databaseName/ =~ o }
+  end
+
+if derby_dir
+  directory ::File.dirname(derby_dir.split('=')[1]) do
```

Comment:
naming-wise, instead of derby_dir, i'd suggest extracting the actual ``derby_db`` including the split('=') above, and then simplifying this line to ``directory ::File.dirname(derby_db) do``.  It would be a lot easier to read.